src/fft.rg

```diff
@@ -13,74 +13,135 @@
 -- limitations under the License.
 
 --[[--
-Regent FFT library.
+Regent FFT library
 ]]
 
 import "regent"
 
+local format = require("std/format")
 local data = require("common/data")
-local cudahelper = require("regent/cudahelper")
+local gpuhelper = require("regent/gpu/helper")
+local gpu_available = gpuhelper.check_gpu_available()
 
+-- Import C and FFTW APIs
 local c = regentlib.c
 local fftw_c = terralib.includec("fftw3.h")
-terralib.linklibrary("libfftw3.so")
+regentlib.linklibrary("libfftw3.so")
+regentlib.linklibrary("libfftw3f.so")
 
-local use_cuda = cudahelper.check_cuda_available()
+-- Import cuFFT API
 local cufft_c
-if use_cuda then
-  cufft_c = terralib.includec("cufft.h")
-  terralib.linklibrary("libcufft.so")
+local cufft_assert
+if gpu_available then
+  cufft_c = terralib.includec("cufftXt.h")
+  regentlib.linklibrary("libcufft.so")
+
+  terra cufft_assert(result : cufft_c.cufftResult)
+    var status = "UNKNOWN"
+    if result == 0 then
```

Comment:
Can we double check if this can be automated? I don't like hard-coding the enum values.

E.g., we might be able to do `cufft_c.CUFFT_SUCCESS`. If that works, then we just need a list of the codes and we can automate the rest of this.

Reply:
Instead of hardcoding, I used tags like `cufft_c.CUFFT_SUCCESS`-- let me know if this is what you were looking for  https://github.com/arjunkunna/regent-fft/commit/7308dd39ae097741e3c4c7d05dbe1d7b69de6a8f

Reply:
Resolved as discussed over email.